### PR TITLE
Different layout for a single extra field

### PR DESCRIPTION
**Pull request by lbalmaceda**

### CODE REVIEW

**Comment by lbalmaceda** [comment]
screen shot 2016-05-06 at 4 31 45 pm

File: lock/src/main/java/com/auth0/android/lock/views/CustomFieldsFormView.java [hunk 1]
```diff
@@ -70,9 +73,56 @@ private void init() {
...
+            emailField.setEnabled(false);
+            fieldContainer.addView(emailField);
+
+            usernameField = new ValidatedInputView(getContext());
+            usernameField.setDataType(ValidatedInputView.DataType.USERNAME);
+            usernameField.setLayoutParams(params);
+            fieldContainer.addView(usernameField);
+        } else {
+            ValidatedInputView emailOrUsernameField = new ValidatedInputView(getContext());
+            emailOrUsernameField.setDataType(ValidatedInputView.DataType.USERNAME_OR_EMAIL);
```
Comment (hzalaz) [inline lock/src/main/java/com/auth0/android/lock/views/CustomFieldsFormView.java]: Why validates both username & email?

File: lock/src/main/java/com/auth0/android/lock/views/CustomFieldsFormView.java [hunk 2]
```diff
@@ -70,9 +73,56 @@ private void init() {
         title = (TextView) findViewById(R.id.com_auth0_lock_title);
         fieldContainer = (LinearLayout) findViewById(R.id.com_auth0_lock_container);
 
+        if (lockWidget.getConfiguration().getExtraSignUpFields().size() == 1) {
```
Comment (hzalaz) [inline lock/src/main/java/com/auth0/android/lock/views/CustomFieldsFormView.java]: Make this condition more readable like `mustUseFieldsFromFirstStep`

File: lock/src/main/java/com/auth0/android/lock/views/CustomFieldsFormView.java [hunk 3]
```diff
@@ -70,9 +73,56 @@ private void init() {
...
+            addDisabledFields();
+        }
         addCustomFields();
     }
 
+    private void addDisabledFields() {
```
Comment (hzalaz) [inline lock/src/main/java/com/auth0/android/lock/views/CustomFieldsFormView.java]: Maybe add a better name to this

File: lock/src/main/java/com/auth0/android/lock/views/CustomFieldsFormView.java [hunk 4]
```diff
@@ -70,9 +78,57 @@ private void init() {
...
         addCustomFields();
     }
 
+    private void addFirstStepFields() {
+        int verticalMargin = (int) getResources().getDimension(R.dimen.com_auth0_lock_widget_vertical_margin_field);
+        LinearLayout.LayoutParams params = new LinearLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT);
+        params.gravity = Gravity.CENTER_HORIZONTAL;
+        params.setMargins(0, verticalMargin / 2, 0, verticalMargin / 2);
+
+        boolean showEmail = false;
```
Comment (hzalaz) [inline lock/src/main/java/com/auth0/android/lock/views/CustomFieldsFormView.java]:
```java
boolean showEmail = false;
boolean showUsername = false;
if (lockWidget.getConfiguration().isUsernameRequired()) {
   showEmail = true;
   showUsername = true;
} else if (lockWidget.getConfiguration().getUsernameStyle() == UsernameStyle.USERNAME) {
   showUsername = true;
} else if (lockWidget.getConfiguration().getUsernameStyle() == UsernameStyle.EMAIL || lockWidget.getConfiguration().getUsernameStyle() == UsernameStyle.DEFAULT) {
   showEmail = true;
}
```
vs
```java
Configuration configuration = lockWidget.getConfiguration();
boolean showEmail = configuration.isUsernameRequired() || configuration.getUsernameStyle() == UsernameStyle.EMAIL || configuration.getUsernameStyle() == UsernameStyle.DEFAULT
boolean showUsername = configuration.isUsernameRequired() || configuration.getUsernameStyle() == UsernameStyle.USERNAME
```

**Pull request merged**